### DISCUSSION TITLE
[Windows] What do I use Flashfetch for?

### DISCUSSION THREAD

**Discussion by ypurpl** [fastfetch-cli/fastfetch#627] [Q&A] [answered]
when installing the scoop binary, there's fastfetch and flashfetch. According to hyperfine, fastfetch is faster

**Comment by CarterLi** [comment, marked as answer]
Always use fastfetch. Flashfetch is meant to be used for compiling from source, and I actually considering removing it from release artifacts because of possible confussion.

**Reply by Antiz96** [reply]
Hi @CarterLi 
I have been asked to stop building `flashfetch` in the Arch Linux package but it's unclear to me what (and when) it is used for.

> Flashfetch is meant to be used for compiling from source

Do you mean it is required at build time when building fastfetch?

From [this comment](https://github.com/fastfetch-cli/fastfetch/issues/777#issuecomment-2109368375):

> Skipping flashfetch build may break a lot of package managers. Closing for now

Could you elaborate on that? Does that mean that skipping `flashfetch` build (by only targeting `fastfetch` with `cmake --build` for instance) could cause issue somehow? I found an [old related comment on the AUR](https://aur.archlinux.org/packages/fastfetch-git?O=20#comment-859498) but it isn't explicit about the issue it caused.

> [...] I actually considering removing it from release artifacts because of possible confussion.

If I understand correctly, you mean removing `flashfetch` from `cmake --install`? If people shouldn't use it, that sounds like a good idea indeed.

To sum up, do I understand correctly that `flashfetch` is required at build time (or rather that skipping its build could create undesired behaviors) but could eventually be omitted at runtime (e.g. do not ship the `flashfetch` binary with the package after it is built)?  
I'm confused about whether I should accept to stop building and / or shipping the `flashfetch` binary within the Arch package or not.

Thanks in advance :)

**Reply by CarterLi** [reply]
Source code of flashfetch: https://github.com/fastfetch-cli/fastfetch/blob/dev/src/flashfetch.c. It compiles the configuration of fastfetch statically in the binary, for people who want to save the short time of config file parsing.

> Could you elaborate on that

A lot packaging script expects fastfetch [ships flashfetch binary](https://github.com/ScoopInstaller/Main/blob/master/bucket/fastfetch.json#L18) for historical reasons. I don't want to break them.

> If people shouldn't use it, that sounds like a good idea indeed.

If they know what they are doing they can use it of course. For people who don't know what flashfetch is, they should always use fastfetch.

**Reply by Antiz96** [reply]
Alright, makes more sense now. Thanks a lot :)

Since the Arch packaging script doesn't necessarily expects the `flashfetch` binary to be shipped and the package itself provides pre-built binaries anyway, I think it makes little to no sense building and shipping `flashfetch` then. Makes sense to drop `flashfetch` from the Arch package right?

**Reply by CarterLi** [reply]
Yeah

**Reply by Antiz96** [reply]
Turns out that not building `flashfetch` creates extra annoyances from a packaging standpoint.  

Targeting specific targets with `cmake build` (to exclude `flashfetch`) instead of simply building everything forces you to be explicit about the targets you want (e.g. `cmake --build build` vs `cmake --build build --target fastfetch --target fastfetch-test-list --target fastfetch-test-format --target fastfetch-test-strbuf`) which is both not convenient and eventually not future proof when you need test scripts (in case one is ever added or removed in the future for instance, which would imply to manually update the target list accordingly).

It also prevents the use of `cmake --install` as it actually expects the presence of `flashfetch` when installing stuff to `DESTDIR`.

I'm not completely familiar with `cmake`, but would a build option such as `-DBUILD_FLASHFETCH='ON'/'OFF'` to "automagically" exclude it from the built targets and ignore it in `cmake_install.cmake` would be doable?

**Reply by Antiz96** [reply]
> I'm not completely familiar with cmake, but would a build option such as -DBUILD_FLASHFETCH='ON'/'OFF' to "automagically" exclude it from the built targets and ignore it in cmake_install.cmake would be doable?

I proposed a patch for the above at https://github.com/fastfetch-cli/fastfetch/pull/1455 :)
